packages/lib-utils/src/app/api-discovery/discovery-cache.ts

```diff
@@ -1,23 +1,53 @@
 import { consoleLogger } from '@monorepo/common';
+import * as _ from 'lodash-es';
+import { getReferenceForModel } from '../../k8s/k8s-utils';
 import type { DiscoveryResources } from '../../types/api-discovery';
+import type { K8sModelCommon } from '../../types/k8s';
 
 const SDK_API_DISCOVERY_RESOURCES_LOCAL_STORAGE_KEY = 'sdk/api-discovery-resources';
 
-export const cacheResources = (resources: DiscoveryResources) => {
+const mergeByKey = (prev: K8sModelCommon[], next: K8sModelCommon[]) =>
+  Object.values(_.merge(_.keyBy(prev, getReferenceForModel), _.keyBy(next, getReferenceForModel)));
+
+const getLocalResources = () => {
+  try {
+    return JSON.parse(localStorage.getItem(SDK_API_DISCOVERY_RESOURCES_LOCAL_STORAGE_KEY) || '{}');
+  } catch (e) {
+    consoleLogger.error('Cannot load cached API resources', e);
+    throw e;
+  }
+};
+
+const setLocalResources = (resources: DiscoveryResources[]) => {
   try {
     localStorage.setItem(SDK_API_DISCOVERY_RESOURCES_LOCAL_STORAGE_KEY, JSON.stringify(resources));
   } catch (e) {
     consoleLogger.error('Error caching API resources in localStorage', e);
     throw e;
   }
 };
+export const cacheResources = (resources: DiscoveryResources[]) => {
+  const allResources = [...[getLocalResources()], ...resources].reduce(
+    (acc, curr) =>
+      _.mergeWith(acc, curr, (first, second) => {
+        if (Array.isArray(first) && first[0]?.constructor?.name === 'Object') {
```

Comment:
Can we replace `first[0]?.constructor?.name === 'Object'` with `_.isPlainObject(first)` ?

Reply:
Not really, the `isPlainObject` will return true for `new String('')`. This one directly returns true only for variables which are key value pair.

Reply:
Actually, testing it will return the same value. So the question is, do we want to use library's function or plain JS one?

Reply:
Lodash implements this more or less the same way (with more checks to see if the constructor is in fact a function): https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L12036-L12047

We've been trying to use ES6/vanilla JS utilities when possible over Lodash in Openshift Console.

@karelhala one thing we may want to consider - if we do not use Lodash's implementation here, perhaps we should introduce a utility function that can be reused that does similar.